bridge/client/app/_components/ktb-create-secret-form/ktb-create-secret-form.component.html

```diff
@@ -9,43 +9,51 @@ <h2>Create Secret</h2>
+            <dt-label>Name *</dt-label>
+            <input formControlName="name" type="text" required dtInput placeholder="Secret name" />
+            <dt-error *ngIf="getFormControl('name')?.errors?.required">Must not be empty</dt-error>
+            <dt-error *ngIf="getFormControl('name')?.errors?.pattern">Name must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character</dt-error>
           </dt-form-field>
         </div>
 
-        <div class="mb-1" fxLayout="row" fxLayoutAlign="none end" fxLayoutGap="15px" *ngFor="let data of secret.data; let index = index;">
-          <dt-form-field>
-            <dt-label>Key</dt-label>
-            <input [(ngModel)]="data.key" [formControlName]="'key'+(index+1)" type="text" required dtInput placeholder="Key" />
-            <dt-error *ngIf="getFormControl('key' + (index+1))?.errors?.required">Must not be empty</dt-error>
-            <dt-error *ngIf="getFormControl('key' + (index+1))?.errors?.pattern">Key must consist of alphanumeric characters, '-', '_' or '.'</dt-error>
-          </dt-form-field>
-          <dt-form-field>
-            <dt-label>Value</dt-label>
-            <input [(ngModel)]="data.value" [formControlName]="'value'+(index+1)" type="password" dtInput placeholder="Value" />
-            <dt-error>Must not be empty</dt-error>
-          </dt-form-field>
-          <button dt-icon-button variant="nested" aria-label="An example nested button containing an icon" title="Remove key-value pair" (click)="removePair(index)">
-            <dt-icon name="removerow"></dt-icon>
-          </button>
+        <div formArrayName="data">
+          <ng-container *ngFor="let dataGroup of data?.controls; let i = index;">
+            <form [formGroup]="dataGroup">
+              <div class="mb-1" fxLayout="row" fxLayoutAlign="none end" fxLayoutGap="15px">
+                <dt-form-field>
+                  <dt-label>Key *</dt-label>
+                  <input formControlName="key" type="text" required dtInput placeholder="Key" autocomplete="false" />
+                  <dt-error *ngIf="dataGroup.get('key')?.errors?.required">Must not be empty</dt-error>
+                  <dt-error *ngIf="dataGroup.get('key')?.errors?.pattern">Key must consist of alphanumeric characters, '-', '_' or '.'</dt-error>
+                </dt-form-field>
+                <dt-form-field>
+                  <dt-label>Value *</dt-label>
+                  <input formControlName="value" type="password" required dtInput placeholder="Value" autocomplete="false" />
+                  <dt-error>Must not be empty</dt-error>
+                </dt-form-field>
+                <button dt-icon-button variant="nested" aria-label="Remove key-value pair" title="Remove key-value pair" (click)="removePair(i)">
```

Comment:
Add `type="button"` here. Else on delete it tries to submit the form but is not possible, which results in this message in the console
![image](https://user-images.githubusercontent.com/11599148/126768663-83f3ea89-33db-4b07-adba-f087c4057672.png)